src/core/lib/channel/channel_args.h

```diff
@@ -281,7 +281,7 @@ class ChannelArgs {
     const grpc_arg_pointer_vtable* vtable_;
   };
 
-  using Value = absl::variant<int, std::string, Pointer>;
+  using Value = absl::variant<int, std::shared_ptr<const std::string>, Pointer>;
```

Comment:
Is there a reasonable way we can use `shared_ptr<>` internally but not expose that in the API?

Reply:
We could not expose `Value` or write a class for it.

Reply:
Ended up needing to wrap it in a class anyway to get equality right.